js/src/util/index.js

```diff
@@ -36,7 +36,23 @@ const getSelector = element => {
   let selector = element.getAttribute('data-bs-target')
 
   if (!selector || selector === '#') {
-    const hrefAttr = element.getAttribute('href')
+    let hrefAttr = element.getAttribute('href')
+
+    if (!hrefAttr) {
+      return null
+    }
+
+    // The only valid content that could double as a selector are ids, so everything starting with . or #
+    // If a 'real' url is used as the selector, document.querySelector will rightfully
+    // complain it is invalid. (see twbs#32273)
+    if (!hrefAttr.includes('#') && !hrefAttr.startsWith('.')) {
```

Comment:
How about merging this with the `!hrefAttr` check?

Reply:
No problem, I thought it was more readable this way.